core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1658,7 +1658,7 @@ class Partition(val topicPartition: TopicPartition,
   def deleteRecordsOnLeader(offset: Long): LogDeleteRecordsResult = inReadLock(leaderIsrUpdateLock) {
     leaderLogIfLocal match {
       case Some(leaderLog) =>
-        if (!leaderLog.config.delete)
+        if (!leaderLog.config.delete && leaderLog.config.compact)
```

Comment:
@junrao WDYT? We have not discussed the support of `DeleteRecordsRequest` in the case where `cleanup.policy` is empty.

Reply:
@m1a2st please update KIP and upgrade.html to match this change.